src/uwtools/config.py

```diff
@@ -752,3 +753,22 @@ def create_config_obj(
                     raise ValueError(err_msg)
                 # Dump to file:
                 dump_method(path=outfile, cfg=config_obj)
+
+
+def print_config_section(config: dict, section_path: List[str], log: Logger) -> None:
+    """
+    Descends into the config via the given section keys, then prints the contents of the located
+    subtree as key=value pairs, one per line.
+    """
+    for key in section_path:
+        if isinstance(config[key], dict):
+            config = config[key]
+        else:
+            log.error(f"{key} type must be a dictionary")
+            raise UWConfigError("Key type must be a dictionary")
+    for key, value in config.items():
+        if type(value) not in (bool, float, int, str):
```

Comment:
Should we handle a list as a value?

Reply:
@christinaholtNOAA Honest question: Would it make sense to export a Python list as the value to assign to a shell environment variable?

Reply:
We have use cases like this in SRW, but it's not immediately clear to me that we _need_ those situations. We might want to consider them in the context of this being a tool we'd use in the interim of having a YAML configuration layer but with some of the drivers still in bash.

One particular case in SRW is in [fixed_files_mapping.yaml](https://github.com/ufs-community/ufs-srweather-app/blob/develop/parm/fixed_files_mapping.yaml), which sets `FV3_NML_VARNAME_TO_SFC_CLIMO_FIELD_MAPPING` to a list that looks sort of like the dictionary we set up for `stage_files`. I'm not sure how many other instances exist.

Reply:
Given something like https://github.com/ufs-community/ufs-srweather-app/blob/develop/parm/fixed_files_mapping.yaml#L158-L169 (if I'm looking at the right thing), what would you expect a bash `export` statement to look like?

Reply:
I think we'd need to end up with a bash iterable so that it might be something like this:

```
 FV3_NML_VARNAME_TO_SFC_CLIMO_FIELD_MAPPING=(
  "FNALBC  | snowfree_albedo" \
  "FNALBC2 | facsf" \
 )
```

Perhaps that doesn't work out well as an iterable, though. Maybe the true iterables won't be able to have spaces in them? So like in lines 86-119?

Reply:
Doing it this way will have edge cases / failure modes that may lead to behavior ranging from mystifying to dangerous depending on user's inputs and how they're using the outputs. If you're ok with that, here's a diff of @elcarpenterNOAA 's current code that does, I think, what you're suggesting:
```
(DEV-uwtools) ~/git/uwtools % git diff -U0
diff --git a/src/uwtools/config.py b/src/uwtools/config.py
index 6ab54d3..71f923f 100644
--- a/src/uwtools/config.py
+++ b/src/uwtools/config.py
@@ -782,0 +783 @@ def print_config_section(config: dict, section_path: List[str], log: Logger) ->
+    isdict = isinstance(config, dict)
@@ -784 +785 @@ def print_config_section(config: dict, section_path: List[str], log: Logger) ->
-        value = config[item] if isinstance(config, dict) else item
+        value = config[item] if isdict else item
@@ -787 +788 @@ def print_config_section(config: dict, section_path: List[str], log: Logger) ->
-        if isinstance(config, dict):
+        if isdict:
@@ -789 +790 @@ def print_config_section(config: dict, section_path: List[str], log: Logger) ->
-    print("\n".join(sorted(output_lines)) if isinstance(config, dict) else f"{keys[-1]}={config}")
+    print("\n".join(sorted(output_lines)) if isdict else "%s=%s" % (keys[-1], " ".join(config)))
```
I added an `isdict` variable to DRY out multiple `isinstance()` calls.

In a REPL:
```
>>> d = {"x": ["a", "b", "c"]}
>>> config.print_config_section(d, ["x"], Logger())
x=a b c
```

Reply:
Not totally sure I understand the question, but in a (working) bash script, the author would already have had to deal with all of the quoting issues to represent the values they want in a form that could be iterated over, possibly including adding parens. There are certainly a lot of mystifying rules/techniques that they would have to deal with, of course, but they would have had to sort all of that out to get to working code. If their values contain spaces, or special characters, or whatever, they would have had to deal with it, and they'd know their specific intent / use case.

The mystery here would come from the fact that a perfectly legit YAML list (for example, one where the values contained spaces), to be used for purposes unknown, would produce something in bash that wouldn't reflect what the YAML said. Like
```
% export 'x="a apple" "b banana"'
% for e in ${x[*]}; do echo $e; done
"a
apple"
"b
banana"
```
Unexpected interpretation of that YAML data.

Something like this might be better:
```
% export x=( "a apple" "b banana" )
% for e in "${x[@]}"; do echo "$e"; done
a apple
b banana
```
Or, bringing `python` into the mix:
```
% cat foo.py 
print('x=( "a apple" "b banana" )')
% eval export $(python foo.py)
% for e in "${x[@]}"; do echo "$e"; done
a apple
b banana
```
I haven't been able to make it work without `eval`, though, and `eval` reasonably makes people nervous. But I don't have a better idea.

Reply:
Well, I do have a better (I think) idea: Use a structured format for structured data, and a tool to destructure it, e.g. JSON + `jq`. We're essentially trying to generate bash code in Python and have bash interpret it, which is tricky, but what we'll have to do if we need a pure-bash solution with list-ish values.

Reply:
Let's go ahead and leave lists out for now. It seems like we're getting into space that is too big a hoop to jump through and we could likely find better ways for the bash scripts to handle passing this information around. Perhaps we can take a harder look at SRW and see if it's better to propose a change on that side.